frontend/src/context/VaultContext.tsx

```diff
@@ -50,8 +40,22 @@ export const VaultContextProvider: React.FC = ({ children }) => {
     if (vaultData && vaultData.user) {
       const userVaults = vaultData.user.vaults as Vault[]
       setVaults(userVaults)
+      if (userVaults.length > 0 && selectedVault === null) {
```

Comment:
nit: can simplify latter half to `!selectedVault`

Reply:
fixed